tools/testing/selftests/bpf/progs/verifier_spill_fill.c

```diff
@@ -938,4 +938,34 @@ l0_%=:	r0 = 0;						\
 	: __clobber_all);
 }
 
+SEC("xdp")
+__description("32-bit fill after 64-bit spill")
+__success __retval(0)
+__naked void fill_32bit_after_spill_64bit(void)
```

Comment:
I think that a second test is needed to check that code below does not break ID relation when spilled reg width is small:

```c
				/* Break the relation on a narrowing fill.
				 * coerce_reg_to_size will adjust the boundaries.
				 */
				if (get_reg_width(reg) > size * BITS_PER_BYTE)
					state->regs[dst_regno].id = 0;
```

Or is it covered by something existing?